### ISSUE TITLE
building zfs against 5.3 fails on module build "No rule to make target 'module/Module.symvers'"

### ISSUE THREAD

**Issue by ColinIanKing** [zfsonlinux/zfs#9133]
Building zfs (today's tip, head commit c81f1790e28a547769ce9d8e2d25391cf4317cc0) against 5.3-rc3 I'm hitting module build failures

### System information

Type | Version/Name
 --- | --- 
Distribution Name	| Ubuntu Eoan
Distribution Version	| 19.10 (Development)
Linux Kernel	|  5.3-rc3 
Architecture	| x86-64 (amd64)
ZFS Version | zfs-0.8.1 (head c81f1790e28a547769ce9d8e2d25391cf4317cc0)
SPL Version	| spl-0.8.1 (head c81f1790e28a547769ce9d8e2d25391cf4317cc0)

Clean build of  today's ZFS tip against linux 5.3-rc3 I'm hitting the following build failure:
```
[lots of build messages ....]
mkdir -p api core spi io os algs algs/aes algs/edonr algs/modes algs/sha1 algs/sha2 algs/skein asm-x86_64 asm-x86_64/aes asm-x86_64/modes asm-x86_64/sha1 asm-x86_64/sha2 asm-i386 asm-generic
make[3]: Leaving directory '/home/cking/zfs/zfs/module/icp'
make[3]: Entering directory '/home/cking/zfs/zfs/module/lua'
mkdir -p setjmp
make[3]: Leaving directory '/home/cking/zfs/zfs/module/lua'
make -C /usr/src/linux-headers-5.3.0-3-generic M=`pwd`  CONFIG_ZFS=m modules
make[3]: Entering directory '/usr/src/linux-headers-5.3.0-3-generic'
  Building modules, stage 2.
make[3]: Leaving directory '/usr/src/linux-headers-5.3.0-3-generic'
make[2]: Leaving directory '/home/cking/zfs/zfs/module'
make[2]: Entering directory '/home/cking/zfs/zfs'
make[2]: *** No rule to make target 'module/Module.symvers', needed by 'all-am'. Stop.
make[2]: Leaving directory '/home/cking/zfs/zfs'
make[1]: *** [Makefile:840: all-recursive] Error 1
make[1]: Leaving directory '/home/cking/zfs/zfs'
make: *** [Makefile:709: all] Error 2

```

To reproduce, I built using:
```
autoconf  -ivf; ./configure; make clean; make -j 8
```

**Comment by ColinIanKing** [comment]
This may be due to the linux headers being incorrect, I will debug this a bit further.

**Comment by jonathonf** [comment]
Hitting the same thing on Manjaro with 5.3rc3.

**Comment by ColinIanKing** [comment]
Not made much progress on this, will report once I find some obvious difference.

**Comment by c0d3z3r0** [comment]
Check if /usr/src/linux-headers-5.3.0-3-generic/Modules.symvers exists, please. It seems this is not a ZFS but a linux issue introduced by this patch https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?h=v5.3-rc3&id=a721588d9475cbbf9e8b3ae1a69b1dea88d01653

**Comment by jonathonf** [comment]
Just for the sake of completeness,

```
$ pacman -Ql | grep -i symvers
linux53-headers /usr/lib/modules/5.3.0-1-MANJARO/build/Module.symvers
```

**Comment by ColinIanKing** [comment]
Reverting commit a721588d9475cbbf9e8b3ae1a69b1dea88d01653 seems to help.  I built today's linux tip with the commit reverted, installed the kernel and can now build zfs tip against it without any issues.

**Comment by c0d3z3r0** [comment]
> Reverting commit a721588d9475cbbf9e8b3ae1a69b1dea88d01653 seems to help. I built today's linux tip with the commit reverted, installed the kernel and can now build zfs tip against it without any issues.

Ah, then my first guess was right after all

**Comment by ColinIanKing** [comment]
I've reported this issue to the author of the commit.

**Comment by philmmanjaro** [comment]
Might be fixed in [this commit](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/scripts/Makefile.modpost?id=47801c97deb71b9e279c15a02a44cf00aa11e7d9). We will see with 5.3-rc4 ...

**Comment by mikoim** [comment]
[v5.3-rc4](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tag/?h=v5.3-rc4) has been tagged, but build still fails.
```
...trimmed...
make -C /home/ek/src/linux M=`pwd`  CONFIG_ZFS=m modules
make[3]: Entering directory '/home/ek/src/linux'
scripts/Makefile.build:57: '/home/ek/src/zfs/module/icp/icp.ko' will not be built even though obj-m is specified.
scripts/Makefile.build:58: You cannot use subdir-y/m to visit a module Makefile. Use obj-y/m instead.
scripts/Makefile.build:57: '/home/ek/src/zfs/module/avl/zavl.ko' will not be built even though obj-m is specified.
scripts/Makefile.build:58: You cannot use subdir-y/m to visit a module Makefile. Use obj-y/m instead.
scripts/Makefile.build:57: '/home/ek/src/zfs/module/nvpair/znvpair.ko' will not be built even though obj-m is specified.
scripts/Makefile.build:57: '/home/ek/src/zfs/module/spl/spl.ko' will not be built even though obj-m is specified.
scripts/Makefile.build:58: You cannot use subdir-y/m to visit a module Makefile. Use obj-y/m instead.
scripts/Makefile.build:58: You cannot use subdir-y/m to visit a module Makefile. Use obj-y/m instead.
scripts/Makefile.build:57: '/home/ek/src/zfs/module/lua/zlua.ko' will not be built even though obj-m is specified.
scripts/Makefile.build:58: You cannot use subdir-y/m to visit a module Makefile. Use obj-y/m instead.
scripts/Makefile.build:57: '/home/ek/src/zfs/module/unicode/zunicode.ko' will not be built even though obj-m is specified.
scripts/Makefile.build:58: You cannot use subdir-y/m to visit a module Makefile. Use obj-y/m instead.
scripts/Makefile.build:57: '/home/ek/src/zfs/module/zcommon/zcommon.ko' will not be built even though obj-m is specified.
scripts/Makefile.build:58: You cannot use subdir-y/m to visit a module Makefile. Use obj-y/m instead.
scripts/Makefile.build:57: '/home/ek/src/zfs/module/zfs/zfs.ko' will not be built even though obj-m is specified.
scripts/Makefile.build:58: You cannot use subdir-y/m to visit a module Makefile. Use obj-y/m instead.
...trimmed...
make[3]: Leaving directory '/home/ek/src/linux'
make[2]: Leaving directory '/home/ek/src/zfs/module'
make[2]: Entering directory '/home/ek/src/zfs'
make[2]: *** No rule to make target 'module/Module.symvers', needed by 'all-am'.  Stop.
make[2]: Leaving directory '/home/ek/src/zfs'
make[1]: *** [Makefile:840: all-recursive] Error 1
make[1]: Leaving directory '/home/ek/src/zfs'
make: *** [Makefile:709: all] Error 2
```

**Comment by ColinIanKing** [comment]
Maybe this will help:

**Comment by jonathonf** [comment]
Works for me! :)

**Comment by behlendorf** [comment]
Resolved in master by commit ff4b68eedc307e6e9b7f3890b809cbb0e9d73856, this fix will be applied to 0.8.2 for linux 5.3 compatibility.
